cloud/ec2_test.go

```diff
@@ -473,8 +455,52 @@ func (s *EC2Suite) TestSpawnHostForTask() {
 	s.EqualValues("instanceType", runInput.InstanceType)
 	s.Equal("my_profile", *runInput.IamInstanceProfile.Arn)
 	s.Equal("evg_auto_evergreen", *runInput.KeyName)
+	s.Require().Len(runInput.NetworkInterfaces, 1)
+	s.True(aws.ToBool(runInput.NetworkInterfaces[0].AssociatePublicIpAddress))
+	s.Require().Len(runInput.NetworkInterfaces[0].Groups, 1)
+	s.Equal("sg-123456", runInput.NetworkInterfaces[0].Groups[0])
+	s.Equal("subnet-123456", aws.ToString(runInput.NetworkInterfaces[0].SubnetId))
+	s.Equal("virtual", *runInput.BlockDeviceMappings[0].VirtualName)
+	s.Equal("device", *runInput.BlockDeviceMappings[0].DeviceName)
+	s.Nil(runInput.SecurityGroupIds)
+	s.Nil(runInput.SecurityGroups)
+	s.Nil(runInput.SubnetId)
+	s.Equal(base64OfSomeUserData, *runInput.UserData)
+}
+
+func (s *EC2Suite) TestSpawnHostForTaskWithoutPublicIPv4Address() {
+	s.h.Distro.Id = "distro_id"
+	s.h.Distro.Provider = evergreen.ProviderNameEc2OnDemand
+	providerSettings := validEC2ProviderSettings()
+	providerSettings.Append(birch.EC.Boolean("is_vpc", true))
+	providerSettings.Append(birch.EC.Boolean("do_not_assign_public_ipv4_address", true))
+	s.h.Distro.ProviderSettingsList = []*birch.Document{providerSettings}
+	s.Require().NoError(s.h.Insert(s.ctx))
+
+	ctx, cancel := context.WithCancel(s.ctx)
+	defer cancel()
```

Comment:
Small nit but do we need a new context or could we use s.ctx in place of it?

Reply:
We can use `s.ctx`. I just forgot about this while I was refactoring the almost-identical-but-not-quite-the-same tests.